SConstruct

```diff
@@ -80,11 +81,15 @@ class FreelanEnvironment(Environment):
             self.install_prefix = self.prefix
             self.bin_install_prefix = self.bin_prefix
 
-        if os.path.basename(self['CXX']) == 'clang++':
-            self.Append(CXXFLAGS=['-Qunused-arguments'])
-            self.Append(CXXFLAGS=['-fcolor-diagnostics'])
-        elif os.path.basename(self['CXX']).startswith('g++'):
-            self.Append(CXXFLAGS=['-Wno-missing-field-initializers'])
+        if self['CXX'] is None:
```

Comment:
You said you are not satisfied with `BuildError` and I tend to agree. In this case what we want is fail with the most meaningful error message possible.

Perhaps a simple `assert` would do ? Something like:

assert 'CXX' in self, "No C++ compiler was detected. Is g++ or clang++ available in your PATH ?"

What do you think ?